### DISCUSSION TITLE
No Data Inbounds while clipping timeseries enabled netcdf

### DISCUSSION THREAD

**Discussion by SarvaPulla** [corteva/rioxarray#251] [Q&A]
Hello,

I am trying to clip GRACE netcdf files using a shapefile. The following workflow works for certain regions/shapefiles: https://gis.stackexchange.com/questions/354782/how-to-mask-netcdf-time-series-data-from-a-shapefile-in-python/354798#354798. But it's intermittent. For example, it didn't work with a missouri shapefile. It keeps throwing a no data found inbounds error for variable 'lwe_thickness'. The data could be NaNs but there definitely is data all over the globe. Here is a sample dataset that I was trying against: https://tethys-staging.byu.edu/thredds/catalog/data/grace/grace/catalog.html?dataset=testDatasetScan/grace/grace/GRC_jpl_gw.nc. There are other datasets here as well: https://tethys-staging.byu.edu/thredds/catalog/data/grace/grace/catalog.html

**Comment by snowman2** [comment]
Are you passing in the CRS of your geometry when clipping?

Here is an example: https://corteva.github.io/rioxarray/stable/examples/clip_geom.html#Clip-using-a-GeoDataFrame

**Reply by SarvaPulla** [reply]
Yes, setting CRS on the shapefile and CRS for the netcdf file.

**Reply by snowman2** [reply]
Great, did you pass in the CRS of the shapefile in when clipping?

**Reply by SarvaPulla** [reply]
Yes. clipped = ds.rio.clip(mo_shp.geometry.apply(mapping), mo_shp.crs, drop=True).

**Reply by SarvaPulla** [reply]
The slight deviation is that I use xarray.open_dataset to read the file instead of xarray.open_dataarray. Was having some other issues when I used open_dataarray. It still fails with open_dataset or open_dataarray.

**Reply by snowman2** [reply]
I would recommend using `rio.clip_box` and see what data is there based on the bounding box for the area of interest.

**Reply by SarvaPulla** [reply]
Clip Box worked on all the shapes that weren't working before. Noticed that the longitude values go from 0 to 360 instead of -180 to 180. Curios about why clip_box worked though.

**Reply by snowman2** [reply]
See #58

**Reply by SarvaPulla** [reply]
Thanks. That helped. Adding the write_crs syntax for reference. Was having issues setting the crs from the example in the docs. The following fixed it.
```
    ds = ds.assign({"lon": (((ds.lon + 180) % 360) - 180)}).sortby('lon')
    ds['lwe_thickness'] = ds['lwe_thickness'].rio.write_crs("epsg:4326")
```
